src/app/file-explorer.js

```diff
@@ -250,10 +235,11 @@ function fileExplorer (appAPI, files) {
     element.parentElement.replaceChild(el, element)
     element = el
   }
-
-  element.events = events
-  element.api = api
-  return element
+  
+  var events = new EventManager()
+  var api = { events: events }
+  container.appendChild(element)
+  return api
```

Comment:
why do you return `api` ?
could you put 
`this.events = new EventManager()`
`container.appendChild(element)`
at the beginning of the file?